### PR TITLE
Restore unbuffer

### PR DESCRIPTION
**Pull request by eddyashton**
Reverts microsoft/CCF#6246.

`unbuffer` was removed in that PR so we could use the Python debugger on the e2e tests. However, it also means we don't get coloured output when running the tests locally. I think the latter is a much bigger loss, and the former is a rare flow where people can manually omit `unbuffer`.

I think the coloured output is a huge readability win:
image
vs
image

Open for discussion!

### CODE REVIEW

**Review by achamayou — APPROVED** [review]

**Comment by eddyashton** [comment]
Ok, AzureLinux can't handle `unbuffer`, so this has to be opt-in on the client side. I don't like recommending changes to your `.bashrc`, but for once I am.

**Review by maxtropets — APPROVED** [review]

**Comment by achamayou** [comment]
@eddyashton is it not just a package we need to install?

**Review by achamayou — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,7 +59,10 @@ function(message)
 endfunction()
 
 option(PROFILE_TESTS "Profile tests" OFF)
-set(PYTHON python3)
+
+# Add CCF_PYTHON_PRELAUNCH=unbuffer to env when calling cmake to get coloured
+# test output
+set(PYTHON "$ENV{CCF_PYTHON_PRELAUNCH}" python3)
```
Comment (maxtropets) [inline CMakeLists.txt:65]: Needs defining in CI yamls to get it to work?

Comment (eddyashton) [reply CMakeLists.txt:65]: I think it's fine if we don't turn this on in the CI. The coloured logs show in some views, but appear as escape-strings in others making it really hard to read/copy.

Comment (achamayou) [reply CMakeLists.txt:65]: There's a $CI we can use though, to have it on locally by default?

**Pull request merged**